source-code/core/src/lint/rules/additionalKey.ts

```diff
@@ -0,0 +1,24 @@
+import type { Context } from '../context.js'
+import { createRule } from '../rule.js'
+
+export const additionalKeyRule = createRule('inlang.additionalKey', () => {
+	let context: Context
+	let referenceLanguage: string
+
+	return {
+		initialize: (config) => {
+			context = config.context
+			referenceLanguage = config.referenceLanguage
+		},
+		visitors: {
+			Resource: ({ target }) => {
+				if (target && target.languageTag.name === referenceLanguage) return 'skip'
+			},
+			Message: ({ target, reference }) => {
+				if (!reference && target) {
+					context.report(target, `Message with id '${target.id.name}' is specified, but missing in the reference`)
```

Comment:
We should probably use objects instead of positional arguments to increase the flexibility to extend APIs like `context.report` in the future. 

```ts
// ❌ can only be extended with more positional arguments
//    every positonal argument must be provided
context.report(target, message, id)
context.report(target, message, undefined, level)

// ✅ mix and match
context.report({ target, message, id })
context.report({ target, message, level })
context.report({ level, target, id })

// ✅ easier deprecation
//    better target is just a dummy name for a property that replaces the fictional old `target`
//    not using positional arguments makes such breaking but not breaking change possible.
context.report({ betterTarget, message })
```

Reply:
It leads to a tiny bit more boilerplate but the tradeoff of 

1. future-proofing the API 
2. (easier comprehension e.g. "ah blabla error is the message" `message: "Blabla error"`)

seems like a good tradeoff to do.

Reply:
Ok, I'll change that